### PR TITLE
KOGITO-6245: Fix issue between quarkus-jsonb & quarkus-resteasy when using runtime-tools-quarkus-extension

### PR DESCRIPTION
**Pull request by pefernan**
Removed `quarkus-jsonb` from the `runtime-tools-quarkus-extension` to avoid conflicts with ` quarkus-resteasy`

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

File: runtime-tools-quarkus-extension-parent/runtime-tools-quarkus-extension/src/main/java/org/kie/kogito/runtime/tools/quarkus/extension/runtime/dataindex/DataIndexService.java [hunk 1]
```diff
@@ -16,45 +16,47 @@
...
 import org.kie.kogito.runtime.tools.quarkus.extension.runtime.forms.FormsStorage;
 
-import io.quarkus.arc.Arc;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
 
 @Path("/dataindex")
 public class DataIndexService {
 
+    private static final ObjectMapper MAPPER = new ObjectMapper();
```
Comment (tiagodolphine) [inline runtime-tools-quarkus-extension-parent/runtime-tools-quarkus-extension/src/main/java/org/kie/kogito/runtime/tools/quarkus/extension/runtime/dataindex/DataIndexService.java]: I think you can `@Inject` the object mapper using CDI, in this way we reuse the default one from quarkus.

**Review by paulovmr — APPROVED** [review]

File: runtime-tools-quarkus-extension-parent/runtime-tools-quarkus-extension/src/main/java/org/kie/kogito/runtime/tools/quarkus/extension/runtime/dataindex/DataIndexService.java (hunk 1 above)
Comment (pefernan) [reply runtime-tools-quarkus-extension-parent/runtime-tools-quarkus-extension/src/main/java/org/kie/kogito/runtime/tools/quarkus/extension/runtime/dataindex/DataIndexService.java]: @tiagodolphine good catch going to give it a try

Comment (pefernan) [reply runtime-tools-quarkus-extension-parent/runtime-tools-quarkus-extension/src/main/java/org/kie/kogito/runtime/tools/quarkus/extension/runtime/dataindex/DataIndexService.java]: @tiagodolphine done

**Comment by manstis** [comment]
FYI @evacchi should we be trying to re-use `InternalObjectMapper` in all kogito extensions?

re: https://github.com/kiegroup/kogito-runtimes/pull/1645/

**Comment by pefernan** [comment]
@manstis I'll wait for @evacchi answer before moving forward

**Comment by manstis** [comment]
@pefernan Sure, no problem from my perspective. @evacchi made the change in `-runtimes` for good reason.

**Review by tiagodolphine — APPROVED** [review]

File: runtime-tools-quarkus-extension-parent/runtime-tools-quarkus-extension/src/test/java/org/kie/kogito/runtime/tools/quarkus/extension/runtime/dataindex/DataIndexServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+import org.kie.kogito.runtime.tools.quarkus.extension.runtime.forms.FormsStorage;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectMock;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.mockito.Mockito.when;
+
+@QuarkusTest
```
Comment (jstastny-cz) [inline runtime-tools-quarkus-extension-parent/runtime-tools-quarkus-extension/src/test/java/org/kie/kogito/runtime/tools/quarkus/extension/runtime/dataindex/DataIndexServiceTest.java]: Does this need to be a quarkus test? This way it's not reflected in the code coverage report (cause it's not well configured for quarkus tests in the kogito-apps repo as whole). Fine for now if rewrite of the test would be complicated.

**Review by jstastny-cz — COMMENTED** [review]
Just one question on the test. SonarCloud coverage is hurt by the fact it's QuarkusTest. The reported gap is actually covered by the test quite well.

**Comment by evacchi** [comment]
the `InternalObjectMapper` to me is meant to be used by the public API
ie. the method `myDataContext.as(SomeOtherDataContext.class)`

now, on Quarkus,`InternalObjectMapper` happens to share Quarkus's ObjectMapper;
however, I don't think we should mix our REST stuff with `InternalObjectMapper`

I *do* believe it would be advisable to share an ObjectMapper for all REST-related matters but then it should be a **separate interface**


now, both interfaces may end up using Quarkus' shared OM, but logically
they should be seprate entitites in other words

The situation may end up being


```
InternalObjectMapper (APIs)
             \
            QuarkusObjectMapper
             /
RestObjectMapper
```

but that is an implementation detail. By keeping the interfaces separate, we won't tangle the two concerns, and we can always decide to have separate implementations if it makes sense.


TBF I am not even sure I want the APIs to use the Quarkus' one it's just an experiment at the moment

So, if you believe that your impl should Quarkus's, probably you should go straight to Quarkus's.

What we _do_ want is to try to have logical concerns of the same kind are grouped together.


- APIs
- REST
- JSONSchema
- Others?

eg.

```
kogito-runtimes/jbpm/jbpm-flow/src/main/java/org/jbpm/process/core/transformation/JsonResolver.java
Line 37 in 3283958
private final ObjectMapper objectMapper;

kogito-runtimes/jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/impl/util/TypeTransformer.java
Line 36 in fdc2739
private ObjectMapper mapper;

kogito-runtimes/jbpm/jbpm-flow/src/main/java/org/jbpm/util/JsonSchemaUtil.java
Line 43 in 22195a1
private static ObjectMapper mapper = new ObjectMapper();

kogito-runtimes/kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/persistence/proto/AbstractProtoGenerator.java
Line 36 in f855f1d
protected final ObjectMapper mapper;
```

**Comment by pefernan** [comment]
@evacchi I think this should fall into the Rest category.
Be aware that JsonSchemaUtil is being used during codegen to generate schemas for tasks/processes and also on runtime to enhance the existing task schema. So it may fall in two categories :P

**Review by jstastny-cz — APPROVED** [review]
LGTM, let's see the sonar coverage results. If it's not reflected even now, we might need to have a look at our configuration.

**Comment by jstastny-cz** [comment]
I think this is safe to merge. Coverage is hurt by POJOs not being fully covered by tests. But it's just setter/getter methods, low risk.

**Comment by pefernan** [comment]
Thanks @jstastny-cz

**Pull request merged**